compose/environment-variables/set-environment-variables.md

```diff
@@ -0,0 +1,188 @@
+redirect_from:
+- /compose/env/
+- /compose/link-env-deprecated/
+---
+
+Environment variables are dealt with by either the Compose file or the CLI. Both have multiple ways you can substitute in or set, your environment variables. This is outlined below. 
+
+## Compose file
+
+### Substitute with an `.env` file
+
+The `.env` file is useful if you have multiple environment variables you need to store.
+
+Below is a simple example: 
+
+```console
+$ cat .env
+TAG=v1.5
+
+$ cat docker-compose.yml
+services:
+  web:
+    image: "webapp:${TAG}"
+```
+
+When you run `docker compose up`, the `web` service defined in the Compose file substitues in the
+image `webapp:v1.5` which was set in the `.env` file. You can verify this with the
+[convert command](../../engine/reference/commandline/compose_convert.md), which prints your resolved application config to the terminal:
+
+```console
+$ docker compose convert
+
+services:
+  web:
+    image: 'webapp:v1.5'
+```
+
+The `.env` file is placed at the base of the project directory where you intend to run the `docker compose` command. You can alternatively call upon the file by either using:
+- The [`--file` option in the CLI](../reference/index.md#use--f-to-specify-name-and-path-of-one-or-more-compose-files) 
+- Setting the [`COMPOSE_FILE` environment variable](envvars.md)
+- Using the [`env_file` attribute in the Compose file](../compose-file/index.md#env_file)
+
+For more information on formatting an environment file, see [Use an environment file](env-file.md).
+
+> **Important**
+>
+>The `.env` file feature only works when you use the `docker compose up` command and does not work with `docker stack deploy`.
+{: .important}
+
+
+### Use the `environment` attribute
+
+You can set environment variables in a service's containers with the
+[`environment` attribute](../compose-file/index.md#environment) in your Compose file. It works in the same way as `docker run -e VARIABLE=VALUE ...`
+
+```yaml
+web:
+  environment:
+    - DEBUG=1
+```
+
+You can choose not to set a value and pass the environment variables from your shell straight through to a
+service's containers. It works in the same way as `docker run -e VARIABLE ...`:
+
+```yaml
+web:
+  environment:
+    - DEBUG
+```
+
+The value of the `DEBUG` variable in the container is taken from the value for the same variable in the shell in which Compose is run.
+
+See [`environment` attribute](../compose-file/index.md#environment) for more information.
+
+### Use the `env_file` attribute
+
+You can pass multiple environment variables from an external file through to
+a service's containers with the [`env_file` option](../compose-file/index.md#env_file). This works in the same way as `docker run --env-file=FILE ...`:
+
+```yaml
+web:
+  env_file:
+    - web-variables.env
+```
+
+> **Note**
+>
+>With this option, environment variables declared in the file cannot then be referenced again separately in the Compose file or used to configure Compose.
```

Comment:
What do you mean here?  That we can't use the value set in those files to define a image name in the compose file, for example such as `image: my-image:${TAG}` ?

Reply:
Yes   
 Or does it not matter coz precedence rules will kick in? Shall I remove it?

Reply:
Why not having something like this:
```suggestion
>With this option, environment variables declared in those files cannot be used as reference in the Compose file or to configure Compose.
```
WDYT @aevesdocker ?